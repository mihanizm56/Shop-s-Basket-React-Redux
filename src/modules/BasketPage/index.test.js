import React from "react"
import { shallow } from "enzyme"
import { BasketPage } from "./index"
import {
  MainBlock,
  Header,
  UserBlock,
  NavigationBlock,
  AdvertisingBlock,
  Footer,
} from "../"

describe("BasketPage test", () => {
  it("Renders correct defaultProps", () => {
    const component = shallow(<BasketPage />)

    expect(component).toMatchSnapshot()
  })

  it("Renders correct defaultProps", () => {
    const component = shallow(
      <BasketPage
        adItems={[
          {
            img: "./img/img1.png",
            description: "Платье-миди с расклешенной юбкой",
            code: "82039-11",
            size: "44",
            color: "синий",
            price: 2450,
            numberOfItems: 1,
            id: 1,
          },
          {
            img: "./img/img2.png",
            description: "Туфли женские украшенные кружевными вставками",
            code: "13524-01",
            size: "38",
            color: "черный",
            price: 2450,
            numberOfItems: 1,
            id: 2,
          },
        ]}
        userName="test_name"
        userItems={[
          {
            img: "./img/img1.png",
            description: "Платье-миди с расклешенной юбкой",
            code: "82039-11",
            size: "44",
            color: "синий",
            price: 2450,
            numberOfItems: 1,
            id: 1,
          },
          {
            img: "./img/img2.png",
            description: "Туфли женские украшенные кружевными вставками",
            code: "13524-01",
            size: "38",
            color: "черный",
            price: 2450,
            numberOfItems: 1,
            id: 2,
          },
        ]}
        increment={() => "test_func"}
        decrement={() => "test_func"}
        deleteItem={() => "test_func"}
        promocode="123"
        discount={false}
        discountValue={2100}
        checkDiscount={() => "test_func"}
      />
    )

    expect(component).toMatchSnapshot()
  })

  it("comes with MainBlock", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(MainBlock).length

    expect(findingComponent).toBe(1)
  })

  it("comes with Header", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(Header).length

    expect(findingComponent).toBe(1)
  })
  it("comes with UserBlock", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(UserBlock).length

    expect(findingComponent).toBe(1)
  })
  it("comes with NavigationBlock", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(NavigationBlock).length

    expect(findingComponent).toBe(1)
  })
  it("comes with AdvertisingBlock", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(AdvertisingBlock).length

    expect(findingComponent).toBe(1)
  })
  it("comes with Footer", () => {
    const component = shallow(<BasketPage />)

    const findingComponent = component.find(Footer).length

    expect(findingComponent).toBe(1)
  })
})
