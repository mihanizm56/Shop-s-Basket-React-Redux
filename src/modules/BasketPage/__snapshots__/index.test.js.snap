// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketPage test Renders correct defaultProps 1`] = `
<styled.div>
  <Header />
  <UserBlock
    numberOfItems={0}
    userName="test_string"
  />
  <NavigationBlock />
  <MainBlock
    checkDiscount={[Function]}
    decrement={[Function]}
    deleteItem={[Function]}
    discount={false}
    discountValue={1800}
    increment={[Function]}
    promocode="test"
    userItems={Array []}
  />
  <AdvertisingBlock
    addList={Array []}
  />
  <Footer />
</styled.div>
`;

exports[`BasketPage test Renders correct defaultProps 2`] = `
<styled.div>
  <Header />
  <UserBlock
    numberOfItems={2}
    userName="test_name"
  />
  <NavigationBlock />
  <MainBlock
    checkDiscount={[Function]}
    decrement={[Function]}
    deleteItem={[Function]}
    discount={false}
    discountValue={2100}
    increment={[Function]}
    promocode="123"
    userItems={
      Array [
        Object {
          "code": "82039-11",
          "color": "синий",
          "description": "Платье-миди с расклешенной юбкой",
          "id": 1,
          "img": "./img/img1.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "44",
        },
        Object {
          "code": "13524-01",
          "color": "черный",
          "description": "Туфли женские украшенные кружевными вставками",
          "id": 2,
          "img": "./img/img2.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "38",
        },
      ]
    }
  />
  <AdvertisingBlock
    addList={
      Array [
        Object {
          "code": "82039-11",
          "color": "синий",
          "description": "Платье-миди с расклешенной юбкой",
          "id": 1,
          "img": "./img/img1.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "44",
        },
        Object {
          "code": "13524-01",
          "color": "черный",
          "description": "Туфли женские украшенные кружевными вставками",
          "id": 2,
          "img": "./img/img2.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "38",
        },
      ]
    }
  />
  <Footer />
</styled.div>
`;

exports[`Youtube Renders correct defaultProps 1`] = `
<styled.div>
  <Header />
  <UserBlock
    numberOfItems={0}
    userName="test_string"
  />
  <NavigationBlock />
  <MainBlock
    checkDiscount={[Function]}
    decrement={[Function]}
    deleteItem={[Function]}
    discount={false}
    discountValue={1800}
    increment={[Function]}
    promocode="test"
    userItems={Array []}
  />
  <AdvertisingBlock
    addList={Array []}
  />
  <Footer />
</styled.div>
`;

exports[`Youtube Renders correct defaultProps 2`] = `
<styled.div>
  <Header />
  <UserBlock
    numberOfItems={2}
    userName="test_name"
  />
  <NavigationBlock />
  <MainBlock
    checkDiscount={[Function]}
    decrement={[Function]}
    deleteItem={[Function]}
    discount={false}
    discountValue={2100}
    increment={[Function]}
    promocode="123"
    userItems={
      Array [
        Object {
          "code": "82039-11",
          "color": "синий",
          "description": "Платье-миди с расклешенной юбкой",
          "id": 1,
          "img": "./img/img1.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "44",
        },
        Object {
          "code": "13524-01",
          "color": "черный",
          "description": "Туфли женские украшенные кружевными вставками",
          "id": 2,
          "img": "./img/img2.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "38",
        },
      ]
    }
  />
  <AdvertisingBlock
    addList={
      Array [
        Object {
          "code": "82039-11",
          "color": "синий",
          "description": "Платье-миди с расклешенной юбкой",
          "id": 1,
          "img": "./img/img1.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "44",
        },
        Object {
          "code": "13524-01",
          "color": "черный",
          "description": "Туфли женские украшенные кружевными вставками",
          "id": 2,
          "img": "./img/img2.png",
          "numberOfItems": 1,
          "price": 2450,
          "size": "38",
        },
      ]
    }
  />
  <Footer />
</styled.div>
`;
